src/System.Diagnostics.TraceSource/src/System/Diagnostics/TraceInternal.cs

```diff
@@ -11,9 +11,21 @@ namespace System.Diagnostics
 {
     internal static class TraceInternal
     {
+        static TraceInternal()
+        {
+            // This is where we override default DebugProvider because we know
```

Comment:
This has rare condition - if one thread is adding/removing indentation and some other thread is just starting the tracing here, some of the indent modifications may get lost. Do we really need to duplicate the state?

Reply:
To fix the race condition:
- I removed the static TraceInternal fields for IndentLevel and IndentSize.
- The original DebugProvider prior to calling TraceInternal's static constructor can keep track of updates to IndentLevel and IndentSize.